lib/pact/utils/metrics.rb

```diff
@@ -0,0 +1,62 @@
+require 'securerandom'
+require 'digest'
+require 'socket'
+require 'pact/version'
+
+module Pact
+  module Utils
+    class Metrics
+
+      def self.report_metric(event, category, action, value = 1)
+
+        if track_events?
+          Pact.configuration.output_stream.puts "WARN: Please note: we are tracking events anonymously to gather important usage statistics like Pact-Ruby version
+            and operating system. To disable tracking, set the 'pact_do_not_track' environment
+            variable to 'true'."
+
+          event = {
+            "v" => 1,
+            "t" => "event",
+            "tid" => "UA-117778936-1",
+            "cid" => calculate_cid,
+            "an" => "Pact Ruby",
+            "av" => Pact::VERSION,
+            "aid" => "pact-ruby",
+            "aip" => 1,
+            "ds" => ENV["CI"] || "unknown",
+            "cd2" => ENV['PACT_EXECUTING_LANGUAGE'] ? "client" : "unknown",
```

Comment:
What is cd2? I think this will always be 'client'.

Reply:
Ah I think what's happened is I have mixed up the 'ds' field and 'cd2' field. 
ds should be one of: client, cli, broker
while cd2 should be "One of: CI, development, unknown"
For cd2 Ron suggested it should be "CI" if the CI env var is set or otherwise "unknown". If that sounds good I'll fix up these 2 fields